lib/src/list.dart

```diff
@@ -20,33 +20,40 @@ import 'dart:async';
 import 'dart:collection' as collection;
 
 import 'collections.dart';
+import 'entity.dart';
 import 'native/realm_core.dart';
 import 'realm_class.dart';
-import 'realm_object.dart';
 import 'results.dart';
 
 /// Instances of this class are live collections and will update as new elements are either
-/// added to or deleted from the Realm that match the underlying query.
+/// added to or deleted to the list or from the Realm.
 ///
-///{@category Realm}
-class RealmList<T extends Object> extends collection.ListBase<T> {
+/// {@category Realm}
+abstract class RealmList<T extends Object> with RealmEntity implements List<T> {
+  /// Gets a value indicating whether this collection is still valid to use.
+  ///
+  /// Indicates whether the [Realm] instance hasn't been closed,
+  /// if it represents a to-many relationship
+  /// and it's parent object hasn't been deleted.
+  bool get isValid;
+
+  factory RealmList._(RealmListHandle handle, Realm realm) => ManagedRealmList._(handle, realm);
+  factory RealmList(Iterable<T> items) => UnmanagedRealmList(items);
+}
+
+class ManagedRealmList<T extends Object> extends collection.ListBase<T> with RealmEntity implements RealmList<T> {
   final RealmListHandle _handle;
-  
-  /// The Realm isntance this collection belongs to.
-  final Realm realm;
 
-  RealmList._(this._handle, this.realm);
+  ManagedRealmList._(this._handle, Realm realm) {
+    setRealm(realm);
+  }
 
-  /// The length of this [RealmList].
   @override
-  int get length => realmCore.getListSize(handle);
+  int get length => realmCore.getListSize(_handle);
 
-  /// @nodoc
-  //settng lenght is no operation
   @override
-  set length(int length) {}
+  set length(int length) {} // no-op for managed lists
```

Comment:
I wonder if we should have this (`settng lenght on managed lists is a no operation`) in the doc comment as well. Then we will need the doc comment itself.

Reply:
Fixed